hoot-core/src/main/cpp/hoot/core/io/OsmApiWriter.cpp

```diff
@@ -177,13 +177,14 @@ void OsmApiWriter::_changesetThreadFunc()
       else
       {
         //  Log the error
-        LOG_ERROR("Thread: " << std::this_thread::get_id() << "\tError uploading changeset: " << id);
+        LOG_ERROR("Error uploading changeset: " << id << "\t" << request->getErrorString());
         //  Split the changeset on conflict errors
         switch (request->getHttpStatus())
         {
         case 400:   //  Placeholder ID is missing or not unique
         case 404:   //  Diff contains elements where the given ID could not be found
         case 409:   //  Conflict, Split the changeset, push both back on the queue
+        case 412:   //  Precondition Failed, Relation with id cannot be saved due to other member
```

Comment:
Here is the new error code.  If a relation has a member that doesn't exist in the database an HTTP `412` code comes back.  So here we split to find the offending element.